db/hints/manager.cc

```diff
@@ -224,7 +224,9 @@ future<> manager::end_point_hints_manager::stop(drain should_drain) noexcept {
         with_lock(file_update_mutex(), [this] {
             if (_hints_store_anchor) {
                 hints_store_ptr tmp = std::exchange(_hints_store_anchor, nullptr);
-                return tmp->shutdown().finally([tmp] {});
+                return tmp->shutdown().then([tmp] {
+                    return tmp->release();
```

Comment:
AFAIU `release()` should not be needed if `shutdown()` completed successfully.
It IS needed otherwise however. Therefore it has to be in the `finally` clause - not `then`.